cinema/tests/test_movie_api.py

```diff
@@ -128,0 +1,126 @@
+from django.test import TestCase
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from cinema.models import Actor, Genre, Movie
+
+
+class MovieApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        drama = Genre.objects.create(
+            name="Drama",
+        )
+        comedy = Genre.objects.create(
+            name="Comedy",
+        )
+        actress = Actor.objects.create(first_name="Kate", last_name="Winslet")
+        movie = Movie.objects.create(
+            title="Titanic",
+            description="Titanic description",
+            duration=123,
+        )
+        movie.genres.add(drama)
+        movie.genres.add(comedy)
+        movie.actors.add(actress)
+
+    def test_get_movies(self):
+        movies = self.client.get("/api/cinema/movies/")
+        titanic = {
+            "title": "Titanic",
+            "description": "Titanic description",
+            "duration": 123,
+            "genres": ["Drama", "Comedy"],
+            "actors": ["Kate Winslet"],
+        }
+        print(movies.data)
+        self.assertEqual(movies.status_code, status.HTTP_200_OK)
+        for field in titanic:
+            self.assertEqual(movies.data[0][field], titanic[field])
+
+    def test_post_movies(self):
+        movies = self.client.post(
+            "/api/cinema/movies/",
+            {
+                "title": "Superman",
+                "description": "Superman description",
+                "duration": 123,
+                "actors": [1],
+                "genres": [1, 2],
+            },
+        )
+        db_movies = Movie.objects.all()
+        self.assertEqual(movies.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(db_movies.count(), 2)
+        self.assertEqual(db_movies.filter(title="Superman").count(), 1)
+
+    def test_post_invalid_movies(self):
+        movies = self.client.post(
+            "/api/cinema/movies/",
+            {
+                "title": "Superman",
+                "description": "Superman description",
+                "duration": 123,
+                "actors": [
+                    {
+                        "id": 3,
+                    }
+                ],
+            },
+        )
+        superman_movies = Movie.objects.filter(title="Superman")
+        self.assertEqual(movies.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(superman_movies.count(), 0)
+
+    def test_get_movie(self):
+        response = self.client.get("/api/cinema/movies/1/")

```

Comment:
Consider using Django's `reverse` function to generate the URL for the API endpoint. This will make the test more robust against changes in the URL patterns.